src/SALib/analyze/sobol.py

```diff
@@ -161,9 +160,9 @@ def create_Si_dict(D, calc_second_order):
                    for k in ('S1', 'S1_conf', 'ST', 'ST_conf'))
 
     if calc_second_order:
-        S['S2'] = np.zeros((D, D))
+        S['S2'] = np.empty((D, D))

```

Comment:
@ConnectedSystems - I'm sure you're aware, but note we've had issues with using `np.empty` in the past because it doesn't necessarily return an array of zero values...  Unless immediately filling the array with values, suggest using `np.zeros`

Reply:
Hi Will,

Yes, this was a conscious decision. All array elements get immediately replaced with `np.nan` (see next line).

I replaced `np.zeros` with `np.empty` as the latter is much faster in producing an array of the desired size.
But perhaps this is an overzealous micro-optimization.

![image](https://user-images.githubusercontent.com/4075136/93176077-0f1c4780-f774-11ea-9108-98fc33ee4cbd.png)

Reply:
I'll replace with `np.full()` as it's perhaps more readable